### PR TITLE
Fixes #5673 - Allow deletion of role with users

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

File: app/helpers/roles_helper.rb [hunk 1]
```diff
@@ -8,4 +8,12 @@ def role_link(role)
       end
     end
   end
+
+  def delete_role_confirmation(role)
+    if role.users.any?
+      _("Warning! This will remove %{name} from %{number} users. are you sure?") % {name: role.name, number: role.users.size}
```
Comment (domcleal) [inline app/helpers/roles_helper.rb]: Use `n_()` here for the pluralised form, i.e. "from %{number} user" (1) and "from %{number} users".

File: test/controllers/roles_controller_test.rb [hunk 2]
```diff
@@ -51,12 +51,12 @@ class RolesControllerTest < ActionController::TestCase
     assert_nil Role.find_by_id(role.id)
   end
 
-  test 'roles in use cannot be destroyed' do
+  test 'roles in use can be destroyed' do
```
Comment (domcleal) [inline test/controllers/roles_controller_test.rb]: Might as well delete this I think, or perhaps change it to a test to check you can't delete built in roles. Your unit test coverage of deletion of roles with users is better I think, and cheaper.

**Review by domcleal — CHANGES_REQUESTED** [review]

File: test/controllers/roles_controller_test.rb (hunk 2 above)
Comment (shlomizadok) [reply test/controllers/roles_controller_test.rb]: Changed to testing built-in role

File: app/helpers/roles_helper.rb (hunk 1 above)
Comment (shlomizadok) [reply app/helpers/roles_helper.rb]: Changed to pluralized form

File: test/models/role_test.rb [hunk 3]
```diff
@@ -36,6 +36,15 @@ class RoleTest < ActiveSupport::TestCase
     role.must_be :valid?
   end
 
+  it "Should delete role who has users" do
```
Comment (dLobatog) [inline test/models/role_test.rb:39]: nitpick - lowercase should. can be fixed on merge. Also `FactoryGirl.build` is preferable for creating the User here, it'll prove the point just the same (and the relationship need not to be in db)

**Review by domcleal — APPROVED** [review]

File: test/models/role_test.rb (hunk 3 above)
Comment (domcleal) [reply test/models/role_test.rb:39]: Fixed the case on merge, but I'd prefer in this case to keep the objects persisted in the DB to ensure we test the whole destroy mechanism (there are a few dependent destroys with cached user roles etc).

**Comment by domcleal** [comment]
Merged as a2c1e8d, thanks @shlomizadok.

**Pull request closed without merging**